tachyon/math/elliptic_curves/bls12/bls12_curve.h

```diff
@@ -22,6 +46,75 @@ class BLS12Curve {
     return f;
   }
 
+  // TODO(chokobole): Leave a comment to help understand readers.
+  template <typename G1AffinePointContainer, typename G2PreparedContainer>
+  static Fp12Ty MultiMillerLoop(const G1AffinePointContainer& a,
+                                const G2PreparedContainer& b) {
+    using EllCoeffs = typename bls12::G2Prepared<Config>::EllCoeffs;
+
+    class Pair {
```

Comment:
Isn't it better to define the class outside of the function? 
Also, the name `Pair` seems a bit ambiguous. I would suggest `G1EllCoeffsPair` or something.

Reply:
Why do you think it's better to move it outside the function ? If we move it, then I agree that `Pair` seems ambiguous. but if we maintain it inside the function, then i think it's okay!